xarray/tests/test_combine.py

```diff
@@ -667,6 +667,28 @@ def test_combine_by_coords(self):
         def test_empty_input(self):
             assert_identical(Dataset(), combine_by_coords([]))
 
+    def test_combine_coords_unnamed_arrays(self):
+        objs = [
+            DataArray([0, 1], dims=("x"), coords=({"x": [0, 1]})),
+            DataArray([2, 3], dims=("x"), coords=({"x": [2, 3]})),
+        ]
+        expected = Dataset({"_": ("x", [0, 1, 2, 3])}, coords={"x": [0, 1, 2, 3]})
+        actual = combine_by_coords(objs)
+        assert_identical(expected, actual)
+
+    def test_combine_coords_mixed_datasets_arrays(self):
+        objs = [
+            DataArray([0, 1], dims=("x"), coords=({"x": [0, 1]})),
+            Dataset({"x": [2, 3]}),
+        ]
+        with raises_regex(ValueError, "without providing an explicit name"):
+            combine_by_coords(objs)
+
+    def test_combine_coords_empty_list(self):
+        expected = Dataset()
+        actual = combine_by_coords([])
```

Comment:
I think this should raise an error

Reply:
I saw a function (`test_empty_input`) above `test_combine_coords_empty_list` that does exactly the same thing. It looks like this function isn't called anywhere, so I think it should be removed, especially if combining an empty list of datasets should raise an error as you suggest.